src/plugins/fs/ext.c

```diff
@@ -59,6 +59,76 @@ static guint32 fs_mode_util[BD_FS_MODE_LAST+1] = {
 
 #define UNUSED __attribute__((unused))
 
+static gint8 compute_percents (guint8 pass_cur, guint8 pass_total, gint val_cur, gint val_total) {
+    gint perc;
+    gint one_pass;
+    /* first get a percentage in the current pass/stage */
+    perc = (val_cur * 100) / val_total;
+
+    /* now map it to the total progress, splitting the stages equally */
+    one_pass = 100 / pass_total;
+    perc = ((pass_cur - 1) * one_pass) + (perc / pass_total);
+
+    return perc;
+}
+
+/**
+ * filter_line_fsck: (skip)
+ * Filter one line - decide what to do with it.
+ *
+ * Returns: Zero or positive number as a percentage, -1 if not a percentage, -2 on an error
+ */
+static gint8 filter_line_fsck (const gchar * line, guint8 total_stages, GError **error) {
+    static GRegex *output_regex = NULL;
+    GMatchInfo *match_info;
+    gint8 perc = -1;
+
+    if (output_regex == NULL) {
+        /* Compile regular expression that matches to e2fsck progress output */
+        output_regex = g_regex_new ("^([0-9][0-9]*) ([0-9][0-9]*) ([0-9][0-9]*) (/.*)", 0, 0, error);
+        if (output_regex == NULL) {
+            return -2;
+        }
+    }
+
+    /* Execute regular expression */
+    if (g_regex_match (output_regex, line, 0, &match_info)) {
+        guint8 stage;
+        gint64 val_cur;
+        gint64 val_total;
+
+        /* The output_regex ensures we have a number in these matches, so we can skip
+         * tests for conversion errors.
+         */
+        stage = (guint8) g_ascii_strtoull(g_match_info_fetch (match_info, 1), (char **)NULL, 10);
+        val_cur = g_ascii_strtoll(g_match_info_fetch (match_info, 2), (char **)NULL, 10);
+        val_total = g_ascii_strtoll(g_match_info_fetch (match_info, 3), (char **)NULL, 10);
```

Comment:
Still some missing spaces between function names and ``(``s in the above and also in a couple of places below. @jtulak, please fix this so that the PR can be merged.

Reply:
OK, I tried to find any other case with a `[a-zA-Z0-9_](` regex in my files and it should be done.